distributed/joblib.py

```diff
@@ -35,5 +35,11 @@ def callback_wrapper():
         future.get = future.result  # monkey patch to achieve AsyncResult API
         return future
 
+    def abort_everything(self, ensure_ready=True):
+        # Tell the executor to cancel any task submitted via this instance
+        # as joblib.Parallel will never access those results.
+        executor = self.executor
+        executor.cancel([Future(k, executor) for k in executor.futures])
```

Comment:
It's an easy change to make.  Usually it's not a big deal because users usually don't manipulate keys directly.

Even here I might suggest that the `DistributedBackend` should explicitly track the futures that it produces from `apply_async` in a separate list.